fiware-region-sanity-tests/tests/fiware_region_without_networks_tests.py

```diff
@@ -43,61 +46,88 @@ def __deploy_instance_helper__(self, instance_name, keypair_name=None, sec_group
         :return: None
         """
 
+        # skip if not all servers were deleted
+        if self.test_world['servers']:
+            self.skipTest("Not all the servers were deleted")
+
         flavor_id = self.nova_operations.get_any_flavor_id()
         self.assertIsNotNone(flavor_id, "Problems retrieving a flavor")
 
         image_id = self.nova_operations.find_image_id_by_name(image_name=BASE_IMAGE_NAME)
         self.assertIsNotNone(image_id, "Problems retrieving the image '{}'".format(BASE_IMAGE_NAME))
 
-        if keypair_name is not None:
-            keypair_value = self.nova_operations.create_keypair(keypair_name)
-            self.test_world['keypair_names'].append(keypair_name)
-
-        security_group_name_list = None
-        if sec_group_name is not None:
-            sec_group_id = self.nova_operations.create_security_group_and_rules(sec_group_name)
-            self.test_world['sec_groups'].append(sec_group_id)
-            security_group_name_list = [sec_group_name]
-
-        server_data = self.nova_operations.launch_instance(instance_name=instance_name, flavor_id=flavor_id,
-                                                           image_id=image_id,
-                                                           metadata=metadata,
-                                                           keypair_name=keypair_name,
-                                                           security_group_name_list=security_group_name_list)
-        self.test_world['servers'].append(server_data['id'])
+        # instance prerequisites
+        try:
+            if keypair_name:
+                self.nova_operations.create_keypair(keypair_name)
+                self.test_world['keypair_names'].append(keypair_name)
+
+            security_group_name_list = None
+            if sec_group_name:
+                sec_group_id = self.nova_operations.create_security_group_and_rules(sec_group_name)
+                self.test_world['sec_groups'].append(sec_group_id)
+                security_group_name_list = [sec_group_name]
+        except ClientException as e:
```

Comment:
Idem, I would like separate try-except loop in order to identify clearly when I have a problem with the keypair and when I have a problem with security groups.